Odis signer refactor

alecps opened the pull request:
### Description

_A few sentences describing the overall effects and goals of the pull request's commits.
What is the current behavior, and what is the updated/expected behavior with this PR?_

### Other changes

_Describe any minor or "drive-by" changes here._

### Tested

_An explanation of how the changes were tested or an explanation as to why they don't need to be._

### Related issues

- Fixes #[issue number here]

### Backwards compatibility

_Brief explanation of why these changes are/are not backwards compatible._

### Documentation

_The set of community facing docs that have been added/modified because of this change_

philbow61 committed:
feat: pass fx adapter api keys to oracles (#10443)

karlb commented on packages/phone-number-privacy/signer/src/common/database/wrappers/account.ts:
```diff
@@ -96,10 +95,12 @@ async function insertRecord(
-    await accounts(db, accountsTable).transacting(trx).insert(data).timeout(config.db.timeout)
+    await tableWithLockForTrx(accounts(db, accountsTable), trx)
+      .insert(data)
+      .timeout(config.db.timeout)
```
I don't think using `tableWithLockForTrx` does anything in this case. According to [the docs](https://knexjs.org/guide/query-builder.html#transacting) it will add locking to select statements by turning them into `SELECT ... FOR UPDATE` statements. But since no select is happening here, this looks like a no-op.

I think the same applies to the change in L77 and other changes in this commit where only insert and update statements happen.

karlb commented on packages/phone-number-privacy/signer/src/common/database/utils.ts:
```diff
@@ -34,6 +34,7 @@ export function countAndThrowDBError<T>(
 export function tableWithLockForTrx(baseQuery: Knex.QueryBuilder, trx?: Knex.Transaction) {
+  //
   if (trx) {
     // Lock relevant database rows for the duration of the transaction
```
As far as I understand [the docs](https://knexjs.org/guide/query-builder.html#transacting), this should say

> Lock relevant database rows for the duration of the transaction *when doing a select*

Other operations might create locks due to their normal operations, but will not be modified by calling this function.

karlb reviewed (commented):
Locking the row in `getPerformedQueryCount` ensures that all other processes will be blocked until this transaction is closed. This prevents users from overdrawing their quota temporarily. This is an opportunity to reduce contention by dropping the lock, since we can live with a temporary quota overdraw.

The quota increase in `incrementQueryCount` will still work fine without the lock, since the increment operation is atomic.

karlb commented:
`getRequestExists` selects a single row from the requests table. I could not find an index on the table. If there really is no index, this operation will result in a full table scan and will get slower proportionally to the table size.

Adding an index would greatly speed up the read at the cost of slowing down writes to the table.

jcortejoso committed:
Fix proxy startup error (#10465)

alecps commented on packages/phone-number-privacy/signer/src/common/database/wrappers/account.ts:
```diff
@@ -63,19 +63,18 @@ async function getAccountExists(
   return meter(
     async () => {
       logger.debug({ account }, 'Incrementing query count')
       if (await getAccountExists(db, accountsTable, account, logger, trx)) {
```
do we need to check if account exists here?

jcortejoso committed:
Dockerfile including whole monorepo (#10472)

ninabarbakadze committed:
chore: add usdcbrl and usdceur to rc1 config (#10425)

martinvol committed:
Fix CLI returning zero required votes on proposals with no tx (#10477)

Fixed

Commit:
fix yarn start

Commit:
Merge branch 'master' of github.com:celo-org/celo-monorepo into alecps/odisRemoveDbLock

Commit:
dek fetcher

Commit:
begin signer refactor

Co-authored-by: Mariano Cortesi <mcortesi@gmail.com>

Commit:
saving progress

alvarof2 committed:
Odis tracing 18 (#10459)

* odis-tracing-node18

* yarn command

* Test again

* Pray

* Pray 2

* Pray 3

* Pray 4

* endpoint traces

* Error trace

* Span in io.ts

* Handle errors - OK

* Fix if

* IO tracing

* Common IO tracing

* Event messages

* Authenticate tracing

* Sign tracing

* Span end

* Span ends

* Controller span

* NodeTracerProvider

* Trace names

* Instrumentations

* No ExpressInstrumentation

* Remove parentSpan

* Request

* Ignore paths

* parentSpan

* Ignore more paths

* knex instrumentation

* Knex instrumentation

* Full node inst + manual

* ignoreIncomingPaths for http inst

* Comments

* No cache build

* Common packages

* TSlint error 1

* TSlint error 2

* TSlint errors

* TSlint errors shorthand

* // tslint:disable-next-line:no-floating-promises

* Fix tests?

mcortesi committed:
nicer meter function

Commit:
fix build

Commit:
Merge branch 'alecps/odisRemoveDbLock' of github.com:celo-org/celo-monorepo into alecps/odisRemoveDbLock

Commit:
Merge branch 'master' of github.com:celo-org/celo-monorepo into alecps/odisRemoveDbLock

Commit:
fix lint and lockfile

Commit:
use new meter

mcortesi committed:
remove use of meter()

mcortesi committed:
Simplify Handler creation

Co-authored-by: Alec Schaefer <alecps@users.noreply.github.com>

mcortesi committed:
cleanup controller and handler creation

mcortesi committed:
Move handlers to a new file

mcortesi committed:
wip: tranform actions into handler functions

mcortesi committed:
Several changes

- only send headers once (send())
- remove sendSuccessful and sendFailure from IO types
- remove action handler, inline init calls in specific handlers

mcortesi committed:
fix domain tests

mcortesi committed:
start removing the domain IOs

mcortesi committed:
Remove pnpQuota IO

mcortesi committed:
Remove pnp sign IO

mcortesi committed:
Disable endpoints on start, not by request

mcortesi committed:
Creates resultHandler, to simplify response management

mcortesi committed:
wip: use resultHandler on pnpSign

Co-authored-by: Alec Schaefer <alecps@users.noreply.github.com>

mcortesi committed:
fix sign handler (without rollback)

mcortesi committed:
change the flow of pnpSign. remove transaction for whole request

mcortesi committed:
todo list

mcortesi committed:
many things

mcortesi committed:
Several Changes

- Removes shouldFailOpen feature
- Clean DB Logic
- Removes LEGACY tables (migrations, logic, tests)
- outer error handler now handles OdisError
- removes block number from e2e pnp test
Co-authored-by: Alec Schaefer <alecps@users.noreply.github.com>

Commit:
remove IO in domains

mcortesi committed:
Create Account and Request Services

mcortesi committed:
Move account & request services to their own files

mcortesi committed:
cleanup returned errors for PNP

mcortesi committed:
Add caching to AccountService with lru-cache

mcortesi committed:
Review Tracing: trace only expensive operations

Commit:
todo list

Commit:
fix domain tests

Commit:
factor out resultHandler

Commit:
fix types

mcortesi committed:
fix traceAsyncFunction

mcortesi committed:
Add tracing to sign function, cleanup

mcortesi committed:
fix error with timeoutHandler, remove catchErrorHandler2

mcortesi committed:
fix unused file

mcortesi committed:
fix combiner compilation issues

mcortesi committed:
fix combiner tests

Commit:
update todo list, bump signer version for load test, remove unused metric

Commit:
check counters and TODOs

Commit:
add back migrations

Commit:
bump signer version

Commit:
migrations

Commit:
bump version

Commit:
switch order of composite primary keys in requests table

Commit:
add cancel, isDuplicate always returns false

Commit:
remove trx, add back getRequestExists

Commit:
remove cancel, fix test

Commit:
(to revert) mock db return values for testing

Commit:
(to revert) mock out full node queries for testing

mcortesi committed:
better load test with rps support

mcortesi committed:
Merge remote-tracking branch 'origin/alecps/odisRelease3.0.0' into alecps/odisRemoveDbLock

# Conflicts:
#	packages/phone-number-privacy/combiner/src/pnp/endpoints/quota/io.ts
#	packages/phone-number-privacy/combiner/src/pnp/endpoints/sign/io.ts
#	packages/phone-number-privacy/signer/package.json
#	packages/phone-number-privacy/signer/src/pnp/endpoints/sign/action.ts
#	yarn.lock
